app/navigators/home-navigator.tsx

```diff
@@ -26,21 +29,38 @@ function UnderConstructionScreen() {
     </View>
   )
 }
-const HomeTabs = createBottomTabNavigator()
-export function HomeNavigator() {
-  const { colors, dark } = useTheme()
 
+interface TabBarLabelProps {
+  focused: boolean
+  children: any
+}
+
+const TabBarLabel: React.FC<TabBarLabelProps> = ({ focused, children }) => {
   return (
-    <SafeAreaView
+    <Text
       style={{
-        backgroundColor: dark ? RneDarkTheme.darkColors.platform.ios.primary : "#fff",
-        height: "100%",
-        width: "100%",
+        color: focused ? palette.black : RneLightTheme.darkColors.grey4,
+        fontSize: 14,
+        paddingTop: 4,
+        fontWeight: "600",
       }}
     >
+      {children}
+    </Text>
+  )
+}
+
+import { initialWindowMetrics } from "react-native-safe-area-context"
+const HomeTabs = createBottomTabNavigator()
+export function HomeNavigator() {
+  const { colors } = useTheme()
+
+  return (
+    <SafeAreaProvider initialMetrics={initialWindowMetrics}>
```

Comment:
@emadbaqeri  Is there any reason you did not use `SafeAreaProvider` in `app-navigator.tsx` to wrap the all screens?

Reply:
Hi @ghorbani-m, thanks for mentioning it. Because we can use as many `SafeAreaProvider`s as we need and the `SafeAreaView` is going to react based on the closest `SafeAreaProvider`. We have nested the navigation stack for the bottom tab and the main pages, I have decided to use another one to make the traversal of finding the closest insets easier for the `SafeAreaView`. I mean it makes it more accurate to way faster for the `SafeAreaView` by doing. this.